### PR TITLE
:bug: fix: surface errors from mutator failures

### PR DESCRIPTION
**Pull request by takirala**
**What this PR does / why we need it**:

Address https://github.com/kubernetes-sigs/cluster-api/pull/7966#issuecomment-1450675662 in our fork first - to make it easy to surface errors in https://github.com/mesosphere/konvoy2/pull/1984

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

File: cmd/clusterctl/client/cluster/mover.go [hunk 1]
```diff
@@ -909,7 +917,11 @@ func (o *objectMover) createTargetObject(nodeToCreate *node, toProxy Proxy, muta
 	}
 
 	for _, mutator := range mutators {
-		mutator(obj)
+		err = mutator(obj)
+		if err != nil {
```
Comment (dkoshkin) [inline cmd/clusterctl/client/cluster/mover.go]: nit: to match the others
```suggestion
		if err := mutator(obj); err != nil {
```

File: cmd/clusterctl/client/cluster/mover.go [hunk 2]
```diff
@@ -657,7 +661,11 @@ func pauseClusterClass(proxy Proxy, n *node, pause bool, mutators ...ResourceMut
 	clusterClass.SetName(n.identity.Name)
 	clusterClass.SetNamespace(n.identity.Namespace)
 	for _, mutator := range mutators {
-		mutator(clusterClass)
+		err = mutator(clusterClass)
+		if err != nil {
```
Comment (dkoshkin) [inline cmd/clusterctl/client/cluster/mover.go]:
```suggestion
		if err := mutator(clusterClass); err != nil {
```

File: cmd/clusterctl/client/cluster/mover.go [hunk 3]
```diff
@@ -628,7 +628,11 @@ func patchCluster(proxy Proxy, n *node, patch client.Patch, mutators ...Resource
 	clusterObj.SetName(n.identity.Name)
 	clusterObj.SetNamespace(n.identity.Namespace)
 	for _, mutator := range mutators {
-		mutator(clusterObj)
+		err = mutator(clusterObj)
+		if err != nil {
```
Comment (dkoshkin) [inline cmd/clusterctl/client/cluster/mover.go]:
```suggestion
		if err := mutator(clusterObj); err != nil {
```

**Comment by dlipovetsky** [comment]
/lgtm

Static checks (linters) ensure that we do not discard an error when we call a mutator.

**Review by dlipovetsky — APPROVED** [review]

**Review by dkoshkin — APPROVED** [review]

**Pull request merged**